pkg/exclusions/exclusions.go

```diff
@@ -0,0 +1,102 @@
+package exclusions
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/google/osv-scanner/internal/sbom"
+	"github.com/google/osv-scanner/pkg/lockfile"
+	"github.com/google/osv-scanner/pkg/reporter"
+)
+
+// ExcludePackages checks for matches in a Lockfile's Packages against the list of provided exclusion regex patterns.
+// If a match is found, the package is removed from the Lockfile's Packages.
+// The updated Lockfile is returned.
+func ExcludePackages(r reporter.Reporter, exclusionPatterns []string, parsedLockfile *lockfile.Lockfile) (*lockfile.Lockfile, error) {
+	// Create a new slice of packages to hold the filtered packages.
+	// Filtered packages are packages that do not match any of the exclusion patterns.
+	var filteredPackages []lockfile.PackageDetails
+
+	// Iterate through the packages in the Lockfile and check for matches against the exclusion patterns.
+	for _, pkg := range parsedLockfile.Packages {
+		matched, err := checkForRegexMatch(r, exclusionPatterns, pkg.Name)
+		if err != nil {
+			return parsedLockfile, err
+		}
+		if !matched {
```

Comment:
Happy path down the left side. Invert this and `continue`.

Then do the append outside of the conditional.

Reply:
Thanks!

Ran with this:
```
func (r *MockReporter) PrintText(text string) {
	r.Output = append(r.Output, text)
}
func (r *MockReporter) PrintError(text string) {
	r.Error = append(r.Output, text)
}
```

Reply:
Would saw the two `if` statements around. Test error first.